README.rst

```diff
@@ -630,6 +630,28 @@ Defaults to ``None``. This option is useful if you want to redirect all emails t
         'OVERRIDE_RECIPIENTS': ['to@example.com', 'to2@example.com']
     }
 
+
+Message-ID
+----------
+
+In order to track and find emails delivered to remote SMTP servers, **django-post_office** generates a unique Message-ID_
+for each email. This identifier then stored with each email and can be looked up in the Django admin backend. Typically
+the Message-ID consists of two parts separated by the ``@`` symbol: The left part is a generated pseudo random number.
+The right part is a constant string, typically denoting the domain name of the sending entity.
+
+It shall be configured using
+
+.. code-block:: python
+
+    # Put this in settings.py
+    POST_OFFICE = {
+        'MESSAGE_ID_RIGHT': 'example.com'
```

Comment:
I think we can auto generate message ID without having this setting. We can look at the domain part of the `from` address and just use that. One less settings to configure and get wrong :)

Reply:
If unset, Django uses the local host name.

The standard states:

> Message-ID consists of a long string of characters (unique identifier of this message on the mail server) ending with the FQDN of the mail server (e.g. 25eb7e48-2252-48ea-80ce-cf2f6119a8e3@ENV95-E2013-1.domain95.lab). Message-ID of an email can be found in the message header. When using CodeTwo software, you can also find this identifier in the software's log files.

Therefore, I wouldn't use the domain part of the `From:` address, because it's not the FQDN of the mail server.
If unconfigured, post_office falls back to the behaviour as Django itself.

Reply:
Ok. If that's the case, let's rename this to `MESSAGE_ID_FQDN` which is much more descriptive. Also, should message ID be generated for all emails, or just those with `MESSAGE_ID_FQDN` set? I propose the latter so only those who need it can turn it on.

For my production use case which has millions of emails in the DB, this will use a fair bit of space, so I'll take all the savings that I can get.

Reply:
The generation of a Message-ID now is optional. It can be activated using a configuration directive.